### PR TITLE
27 icons

### LINKED ISSUE #27: Implement visualisation activity
Implement that the icon shows
whenever the extension is active

### PR DESCRIPTION
**Pull request by thervh70**
Will close #27.

This feature includes visuals! As well as a styling of the popup.

### CODE REVIEW

**Comment by mpsijm** [comment]
About the images file(name)s: 
1. Maybe you could put them in a separate folder, like `img/`
2. I'd like it if the filenames, like in other chrome extensions, have their size in it. So `icon_tiny_off.png` would be renamed to `icon_off19.png`

File: resources/popup.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+$(document).ready(function () {
```
Comment (mdingjan) [inline resources/popup.js]: This statement doesn't have to include the two functions, right?

Comment (thervh70) [inline resources/popup.js]: (jQuery Documentation)[https://learn.jquery.com/using-jquery-core/document-ready/] shows that using this line the script is activated when the document is loaded. This way the script can be placed in the header of the html page

**Comment by thervh70** [comment]
@mpsijm done!

File: resources/manifest.json [hunk 2]
```diff
@@ -7,9 +7,9 @@
   "options_page": "options.html",
 
   "icons": {
-    "16": "icon_off.png",
-    "48": "icon_off.png",
-    "128": "icon_off.png"
+    "16": "img/icon/icon_off.png",
+    "48": "img/icon/icon_off.png",
+    "128": "img/icon/icon_off.png"
```
Comment (mpsijm) [inline resources/manifest.json]: Yes, I'm going to be picky again :P The URL now has the word `icon` twice now, so I'd like `img/icon/off.png` better :)

File: resources/popup.js (hunk 1 above)
Comment (mpsijm) [inline resources/popup.js]: I think @mdingjan means the two functions `read_status` and `set_popup`. I think it doesn't matter where these functions are, being inside the scope of the `ready` callback only means that they can only be used inside that callback, so in this case they don't pollute the global scope :)

File: src/octopeer-github/main/MainController.ts [hunk 3]
```diff
@@ -48,6 +54,12 @@ class MainController {
             });
             sendResponse({});
         });
+        chrome.alarms.create({periodInMinutes: 0.02});
+        let i = 0;
+        chrome.alarms.onAlarm.addListener(function() {
+            i = (i + 1) % 4;
+            self.status.set(i);
+        });
```
Comment (mpsijm) [inline src/octopeer-github/main/MainController.ts]: Maybe add a TODO on this alarm, as this is going to be fixed in #46 :)

File: src/octopeer-github/main/Status.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+            "img/icon/icon_off.png",
+            "img/icon/icon_standby.png",
+        ];
+
+        const tiny = [
+            "img/icon/icon_error19.png",
+            "img/icon/icon_running19.png",
+            "img/icon/icon_off19.png",
+            "img/icon/icon_standby19.png",
+        ];
```
Comment (mpsijm) [inline src/octopeer-github/main/Status.ts]: My refactoring fingers are itching XD How about:
```
private getIcon(status: StatusCode = StatusCode.OFF, size: number = 0) {
    const iconname = [ // can be moved outside scope, as StatusCode is global as well
        "error",
        "running",
        "off",
        "standby",
    ];
    let sizestring = "";
    switch (size) {
        case 19: // can have multiple (valid) cases later, like 16, 38, 48 or 128
            sizestring = `${size}`;
            break;
        default: break;
    }
    return `img/icon/icon_${iconname[status]}${sizestring}.png`
}

File: src/octopeer-github/main/Status.ts [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+        chrome.storage.local.set({
+            status: status,
+        });
+
+        chrome.runtime.sendMessage({line: text[status], path: icon[status]});
+        chrome.browserAction.setIcon({path: tiny[status]}, function() {
+            if (chrome.runtime.lastError) {
+                console.log(chrome.runtime.lastError.message);
+            }
```
Comment (mpsijm) [inline src/octopeer-github/main/Status.ts]: This callback function can now be removed. It was used for debugging (by me :innocent: ) when we couldn't get the icons to work :)

File: resources/popup.js (hunk 1 above)
Comment (thervh70) [inline resources/popup.js]: fixed!

Comment (mdingjan) [inline resources/popup.js]: Yes, Maarten correctly explained my remark. ;)

File: README.md [hunk 6]
```diff
@@ -72,3 +72,6 @@ The extension is now loaded in Chrome.
 ### Adding new type definitions for dependencies
  - For every new dependency, run `typings install --ambient --save <package-name>`
  - The type definitons should now automatically be included via the TypeScript configuration files
+
+### Acknowledgments
+We would like to thank [PixelKit](http://pixelkit.com/) for allowing us to use and modify the lightbulb icon
```
Comment (mdingjan) [inline README.md]: Please add the rights that are applicable to the icon (as Bastiaan said this morning).

**Comment by mpsijm** [comment]
Okay, I'm happy with this branch now :) can be merged once @mdingjan's final comment about the licence is fixed :D

File: src/octopeer-github/main/Status.ts [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+        chrome.storage.local.set({
+            status: status,
+        });
+
+        chrome.runtime.sendMessage({line: Status.MESSAGE[status], path: this.getIcon(status)});
+        chrome.browserAction.setIcon({path: this.getIcon(status, 19)});
+    }
+
+    /**
+     * @param status        the StatusCode to get the icon for.
```
Comment (mdingjan) [inline src/octopeer-github/main/Status.ts]: I know the function of getIcon is clear based on just the name, but please give a short description of what getIcon does. 
The rest of the JSDoc in the document is fine. :)

**Comment by mdingjan** [comment]
I've left a few comments which may require a minor change.
When you've fixed these, I'm happy. :)
The icons are nice :D

Notify me when you're done, then I will merge this branch.

File: src/octopeer-github/main/Status.ts (hunk 7 above)
Comment (mpsijm) [inline src/octopeer-github/main/Status.ts]: What it does is "return the relative icon path from the root of the extension", it says so in the `@return` :P

Comment (mdingjan) [inline src/octopeer-github/main/Status.ts]: I know, I know xD
But a formal documentation starts off with a descriptive line and then you define the parameters and the return type (A).

**Pull request merged**